modin/core/execution/client/query_compiler.py

```diff
@@ -0,0 +1,451 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Module contains ``ClientQueryCompiler`` class."""
+
+import pandas
+from pandas._libs.lib import no_default, NoDefault
+from pandas.api.types import is_list_like
+from typing import Any
+from uuid import UUID
+
+from modin.core.storage_formats.base.query_compiler import BaseQueryCompiler
+from modin.utils import _inherit_docstrings
+
+
+@_inherit_docstrings(BaseQueryCompiler)
+class ClientQueryCompiler(BaseQueryCompiler):
+    """
+    Query compiler for sending queries to a remote server.
+
+    This class translates the query compiler API to function calls on a service
+    object, which may be a remote service.
+
+    Parameters
+    ----------
+    id : UUID
+        ID of this query compiler.
+    """
+
+    lazy_execution: bool = True
+
+    def __init__(self, id: UUID):
+        self._id = id
+
+    @classmethod
+    def set_server_connection(cls, conn: Any):
```

Comment:
Why is this a class method? Once we set the server connection we want every instance of a new ClientQueryCompiler to connect to this specific service? Is there ever a case where we don't want this? (E.g. cases where we have multiple services perhaps)